File: python/credit_card_validator.py
def valid_length(card_number):
	if len(card_number) != 16:
		return False
	for ch in card_number:
		if not ch.isdigit():
			return False
	return True
def card_type(card_number):
	if card_number.startswith("4"):
		return "VISA"
	elif card_number.startswith("5"):
		return "MasterCard"
	elif card_number.startswith("37"):
		return "American Express"
	elif card_number.startswith("6"):
		return "Discover"
	else:
		return "Invalid"
def check(card_number):
	total = 0
	double_digit = False
	for i in range(len(card_number) - 1, -1, -1):
		digit = int(card_number[i])
		if double_digit:
			digit *= 2
		total += digit
		double_digit = not double_digit
	return total % 10 == 0
def main():
	card_number = input("Hello, kindly enter card details to verify: ")
	print("Credit card type:", card_type(card_number))

File: javascript/CreditCardValidator.js
const prompt = require('prompt-sync')();

function ValidLength(cardNumber) {
  if (cardNumber.length !== 16) {
    return false;
  }
for (let i = 0; i < cardNumber.length; i++) {
    let ch = cardNumber[i];
    if (ch < '0' || ch > '9') {
      return false;
    }
  }
  return true;
}
function CardType(cardNumber) {
  if (cardNumber.startsWith('4')) {
    return "VISA";
  } else if (cardNumber.startsWith('5')) {
    return "MasterCard";
  } else if (cardNumber.startsWith('37')) {
    return "American Express";
  } else if (cardNumber.startsWith('6')) {
    return "Discover";
  } else {
    return "invalid";
  }
}
function check(cardNumber) {
  let total = 0;
  let shouldDouble = false;

  for (let i = cardNumber.length - 1; i >= 0; i--) {
    let digit = parseInt(cardNumber[i]);
    if (shouldDouble) {
      digit *= 2;
    }
    total += digit;
    shouldDouble = !shouldDouble;
  }

  return total % 10 === 0;
}